simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/PayloadFactory.java

```diff
@@ -30,12 +30,11 @@ AuthorizeRequest createAuthorizeRequest(String tokenId, List<Integer> evseIds) {
         return payload;
     }
 
-    StatusNotificationRequest createStatusNotification(int evseId, int connectorId, ConnectorStatus connectorStatus, Instant currentTime) {
+    StatusNotificationRequest createStatusNotification(int evseId, int connectorId, CableStatus cableStatus, Instant currentTime) {
         StatusNotificationRequest payload = new StatusNotificationRequest();
         payload.setEvseId(evseId);
         payload.setConnectorId(connectorId);
-
-        if (connectorStatus == ConnectorStatus.UNPLUGGED) {
+        if (cableStatus == CableStatus.UNPLUGGED) {
```

Comment:
Is this a temporary logic? I think we agreed on having ConnectorStatus stored somewhere and enrich StatusNotificationRequest by taking it from there.
ConnectorStatus change would be triggered by CableStatus change and by ChangeAvailabilityRequest

Reply:
Its think it should be refactored also at some point. This code was there already.